### ISSUE TITLE
the following codes in `openssl/ssl.h` are related to SRTP, but not under control of macro `OPENSSL_NO_SRTP`

### ISSUE THREAD

**Issue by HeYoushang** [openssl/openssl#7976]
https://github.com/openssl/openssl/blob/1f483a69bce11c940309edc437eee6e32294d5f2/include/openssl/ssl.h#L222
the following codes in `openssl/ssl.h` are related to SRTP, but not under control of macro `OPENSSL_NO_SRTP`. other codes related to SRTP are all under control except here, please help check if we should add OPENSSL_NO_SRTP to take in the following codes, thank you.
```
/* SRTP protection profiles for use with the use_srtp extension (RFC 5764)*/
typedef struct srtp_protection_profile_st {
    const char *name;
    unsigned long id;
} SRTP_PROTECTION_PROFILE;

DEFINE_STACK_OF(SRTP_PROTECTION_PROFILE)
```

**Comment by mattcaswell** [comment]
`OPENSSL_NO_SRTP` removes SRTP functionality from OpenSSL. It ensures that SRTP cannot be used in OpenSSL. It doesn't mean that every last piece of code is excluded. I don't think a change is required here.

**Comment by HeYoushang** [comment]
Yeah, sometimes we don't exclude every last piece of code when removing a functionality. but here comes a pc-lint error reporting that SRTP_PROTECTION_PROFILE is defined not used. 
In most situations, it is better to exclude every piece of code when removing a functionality. 
Or at least, if a functionality is new developed which is included by a macro, we probably will include every piece of code by the macro.
I think it is better to exclude the code here by OPENSSL_NO_SRTP. Otherwise we may need to face the pc-lint error always.
What do you think

**Comment by mattcaswell** [comment]
That seems like an over zealous lint checker to me. There must surely be lots of things defined in header files that aren't used by applications.

**Comment by richsalz** [comment]
This is a little thing that the team probably wouldn't want to spend time on.  But as a code-cleanup contribution (i.e., someone else writes the PR) I think your suggestions make sense.

**Comment by HeYoushang** [comment]
ok, got it, thank you very much

**Comment by paulidale** [comment]
DEFINE_STACK_OF might be worth considering wrapping with a condition -- it produces a number of inline functions.

**Comment by HeYoushang** [comment]
hmm, yeah. I find that `openssl/srtp.h`  also includes `openssl/ssl.h` (looks like include each other), which includes `openssl/safestack.h` at its first several lines, so it would be problem if moving the above definition into `openssl/srtp.h`. 
And if it really need to use `DEFINE_STACK_OF`  to define something, it shouldn't be problem to include `openssl/safestack.h` (via `openssl/ssl.h`) in `openssl/srtp.h` , should it ?

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
